lib/core/src/Cardano/Wallet/DB/Store/Wallets/Store.hs

```diff
@@ -0,0 +1,122 @@
+    ( EntityField (TxMetaWalletId), TxMeta )
+import Cardano.Wallet.DB.Store.Meta.Model
+    ( DeltaTxMetaHistory (Manipulate), TxMetaHistory, mkTxMetaHistory )
+import Cardano.Wallet.DB.Store.Meta.Store
+    ( mkStoreMetaTransactions )
+import Cardano.Wallet.DB.Store.Transactions.Model
+    ( DeltaTxHistory (..), TxHistoryF (..), mkTxHistory )
+import Cardano.Wallet.DB.Store.Transactions.Store
+    ( mkStoreTransactions )
+import Cardano.Wallet.DB.Store.Wallets.Model
+    ( DeltaTxWalletsHistory (..), walletsLinkedTransactions )
+import Control.Applicative
+    ( liftA2 )
+import Control.Exception
+    ( SomeException )
+import Control.Monad
+    ( forM, forM_ )
+import Control.Monad.Except
+    ( ExceptT (ExceptT), runExceptT )
+import Data.DBVar
+    ( Store (..) )
+import Data.DeltaMap
+    ( DeltaMap (..) )
+import Data.Generics.Internal.VL
+    ( view )
+import Data.List
+    ( nub )
+import Database.Persist.Sql
+    ( SqlPersistT, deleteWhere, entityVal, selectList, (==.) )
+
+import qualified Cardano.Wallet.DB.Store.Meta.Model as TxMetaStore
+import qualified Cardano.Wallet.DB.Store.Transactions.Model as TxStore
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Data.Map.Strict as Map
+
+-- | Store for 'WalletsMeta' of multiple different wallets.
+mkStoreWalletsMeta :: Store
+        (SqlPersistT IO)
+        (DeltaMap W.WalletId DeltaTxMetaHistory)
+mkStoreWalletsMeta =
+    Store
+    { loadS = load
+    , writeS = write
+    , updateS = update
+    }
+  where
+    write reset = forM_ (Map.assocs reset) $ \(wid,metas)
+        -> writeS (mkStoreMetaTransactions wid) metas
+    update _ (Insert wid metas) =
+        writeS (mkStoreMetaTransactions wid) metas
+    update _ (Delete wid) = do
+        deleteWhere [TxMetaWalletId ==. wid ]
+    update _ (Adjust wid da) =
+        updateS (mkStoreMetaTransactions wid) undefined da
+    load :: SqlPersistT
+            IO
+            (Either SomeException (Map.Map W.WalletId TxMetaHistory))
+    load = do
+        wids <- nub . fmap (view #txMetaWalletId . entityVal)
+            <$> selectList @TxMeta [] []
+        runExceptT $ do
+            xs <- forM wids $ ExceptT . loadS . mkStoreMetaTransactions
+            pure $ Map.fromList $ zip wids xs
+
+mkStoreTxWalletsHistory
+    :: Store (SqlPersistT IO) DeltaTxWalletsHistory
+mkStoreTxWalletsHistory =
+    Store
+    { loadS =
+            liftA2 (,) <$> loadS mkStoreTransactions <*> loadS mkStoreWalletsMeta
+    , writeS = \(txHistory,txMetaHistory) -> do
+            writeS mkStoreTransactions txHistory
+            writeS mkStoreWalletsMeta txMetaHistory
+    , updateS = \(txh@(TxHistoryF mtxh),mtxmh) -> \case
+            ExpandTxWalletsHistory wid cs -> do
+                updateS mkStoreTransactions txh
+                    $ TxStore.Append
+                    $ mkTxHistory
+                    $ fst <$> cs
+                updateS mkStoreWalletsMeta mtxmh
+                    $ Adjust wid
```

Comment:
The store and the pure model diverge a little bit here — the function `Wallets.Model.apply` uses `Map.lookup wid mtxmh` to check whether the wallet id is in the map, and uses `Insert` (instead of `Adjust`) if the wallet id is not in the map. 🤔  This works, because of this thing where the database does not distinguish between an empty table and no wallet.

For the sake of consistency: Do you want to exactly mirror the definition of `apply` on the case `ExpandTxWalletsHistory` here, including `Map.lookup`? This would make it visually more obvious that `updateS` and `apply` are equivalent. — If you don't want to do that, I would recommend adding a comment that explains the "thing where the database does not distinguish between an empty table and no wallet" thing for future readers.